src/connections/api.js

```diff
@@ -0,0 +1,193 @@
+require('dotenv').config(); // Load environment variables from .env
+const { onTwitchMessage } = require('../modules/legacy');
+const Logger = require('../utils/logger');
+const WebSocket = require('ws');
+const ReconnectingWebSocket = require('reconnecting-websocket');
+const jwt = require('jsonwebtoken');
+const NodeCache = require('node-cache'); // Import node-cache
+
+class API {
+  #ws;
+  #logger;
+  #pingInterval;
+  #tokenCache; // Token cache
+
+  constructor(wsUrl, wsKey, secretKey, controller) { 
+    this.wsUrl = `${wsUrl}?token=${wsKey}`; // Append wsKey to the WebSocket URL
+    this.secretKey = secretKey || 'your-secret-key';
+    this.controller = controller;
+    this.#logger = new Logger("api");
+
+    // Initialize token cache with a time-to-live of 30 days
+    this.#tokenCache = new NodeCache({ stdTTL: 30 * 24 * 60 * 60 }); // 30 days in seconds
+
+    if (!this.wsUrl) {
+      this.#logger.error('PUBLIC_WS_URL is not defined in environment variables.');
+      process.exit(1);
+    }
```

Comment:
I think this check would need to be moved to the initiator function to do anything. The wsUrl will always be a non-empty string here.

Reply:
Good point, moved the check to the before instantiation